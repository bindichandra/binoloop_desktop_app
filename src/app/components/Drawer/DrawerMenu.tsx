import * as React from "react";
import Box from "@mui/material/Box";
import Drawer from "@mui/material/Drawer";
import Button from "@mui/material/Button";
import List from "@mui/material/List";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import Image from "next/image";

const icons = [
  { src: "/binoloop.svg", alt: "", priority: true },
  { src: "/home.svg", alt: "home", priority: false },
  { src: "/folderOpen.svg", alt: "folderOpen", priority: false },
  { src: "/folder.svg", alt: "folder", priority: false },
  { src: "/plus.svg", alt: "plus", priority: false },
];

interface DrawerMenuProps {
  open: boolean;
  setOpen: (open: boolean) => void;
}

const DrawerMenu: React.FC<DrawerMenuProps> = ({ open, setOpen }) => {
  const toggleDrawer = () => {
    setOpen(!open);
  };

  return (
    <div>
      {/* Render Binoloop icon if the drawer is closed */}
      {!open && (
        <Button onClick={toggleDrawer} className="tw-mt-2">
          <Image src="/binoloop.svg" alt="Binoloop" width={40} height={40} />
        </Button>
      )}

      {/* Drawer */}
      <Drawer
        anchor="left"
        open={true}
        onClose={toggleDrawer}
        variant="persistent"
        sx={{
          width: 100,
          flexShrink: 0,
          "& .MuiDrawer-paper": {
            width: 100,
            boxSizing: "border-box",
          },
        }}
      >
        <Box
          role="presentation"
          onClick={toggleDrawer}
          onKeyDown={toggleDrawer}
        >
          <List>
            {icons.map((icon, index) => (
              <React.Fragment key={index}>
                <div className="tw-flex tw-items-center">
                  <ListItemButton>
                    <ListItemIcon>
                      <Image
                        src={icon.src}
                        alt={icon.alt}
                        width={60}
                        height={60}
                        priority={icon.priority}
                      />
                    </ListItemIcon>
                  </ListItemButton>
                </div>
              </React.Fragment>
            ))}
          </List>
        </Box>
      </Drawer>
    </div>
  );
};

export default DrawerMenu;
