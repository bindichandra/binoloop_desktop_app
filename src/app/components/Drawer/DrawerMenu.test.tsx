import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import DrawerMenu from "./DrawerMenu";

describe("DrawerMenu component", () => {
  test("renders correctly", () => {
    const { container } = render(
      <DrawerMenu open={false} setOpen={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
