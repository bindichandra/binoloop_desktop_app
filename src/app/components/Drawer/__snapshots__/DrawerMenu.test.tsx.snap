// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu component renders correctly 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary tw-mt-2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <img
        alt="Binoloop"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/binoloop.svg"
        style="color: transparent;"
        width="40"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiDrawer-root MuiDrawer-docked css-a2rad9-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="MuiBox-root css-0"
          role="presentation"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <div
              class="tw-flex tw-items-center"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <img
                    alt=""
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="60"
                    src="/binoloop.svg"
                    style="color: transparent;"
                    width="60"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-items-center"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <img
                    alt="home"
                    data-nimg="1"
                    decoding="async"
                    height="60"
                    loading="lazy"
                    src="/home.svg"
                    style="color: transparent;"
                    width="60"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-items-center"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <img
                    alt="folderOpen"
                    data-nimg="1"
                    decoding="async"
                    height="60"
                    loading="lazy"
                    src="/folderOpen.svg"
                    style="color: transparent;"
                    width="60"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-items-center"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <img
                    alt="folder"
                    data-nimg="1"
                    decoding="async"
                    height="60"
                    loading="lazy"
                    src="/folder.svg"
                    style="color: transparent;"
                    width="60"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-items-center"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <img
                    alt="plus"
                    data-nimg="1"
                    decoding="async"
                    height="60"
                    loading="lazy"
                    src="/plus.svg"
                    style="color: transparent;"
                    width="60"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
